aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1893,6 +1951,145 @@ std::tuple<Tensor, Tensor> _triangular_solve_helper_cuda(const Tensor& self, con
   return std::tuple<Tensor, Tensor>(self_working_copy, A_working_copy);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ orgqr ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/*
+  The orgqr function allows reconstruction of an orthogonal (or unitary) matrix Q,
+  from a sequence of elementary reflectors, such as produced by the geqrf function.
+
+  Args:
+  * `self` - Tensor with the directions of the elementary reflectors below the diagonal,
+              it will be overwritten with the result
+  * `tau` - Tensor containing the magnitudes of the elementary reflectors
+  * `infos` - Tensor to store MAGMA's error codes (not used here)
+  * `n_columns` - The number of columns of Q to be computed
+
+  For further details, please see the LAPACK documentation for ORGQR/UNGQR, LARFT, LARFB.
+
+  MAGMA doesn't have native GPU UNGQR function implemented, but it has LARFT, which is used here.
+  The restriction is that it applies only to inputs with tau.shape[-1] <= 32, for other inputs cuSOLVER should be used.
+*/
+template <typename scalar_t>
+inline void apply_orgqr_batched_magma(Tensor& self, const Tensor& tau, Tensor& infos, int64_t n_columns) {
+#ifndef USE_MAGMA
+  TORCH_CHECK(false, "Calling torch.orgqr on a CUDA tensor requires compiling ",
```

Comment:
I thought cuSOLVER could be used if MAGMA wasn't available?

Reply:
That's true. But if cuSOLVER is not available (currently it's disabled for cuda version <  10.1.243) then this function will be called and if MAGMA is also not available this error is raised.